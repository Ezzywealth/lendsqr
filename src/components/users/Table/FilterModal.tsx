// Modal.js
import React from 'react';

import TableHook from './Hook/TableHook';

type Props = {
	setShowFilterModal: (showFilterModal: boolean) => void;
};

const Modal = ({ setShowFilterModal }: Props) => {
	const { onClose, setIsOpen, organization, setOrganization, username, setUsername, phoneNumber, setPhoneNumber, date, setDate, email, setEmail, status, setStatus, applyFilters } = TableHook();

	const handleApply = () => {
		onClose();
		setShowFilterModal(false);
		applyFilters();
	};

	const closeModal = () => {
		setShowFilterModal(false);
		setIsOpen(false);
	};

	return (
		<div className='modal show'>
			<div className='modal-content'>
				<label>
					Organization
					<select value={organization} onChange={(e) => setOrganization(e.target.value)}>
						<option value='' disabled>
							Select
						</option>
						<option className='options' value='lendsqr'>
							lendsqr
						</option>
						<option className='options' value='lendstar'>
							lendstar
						</option>
						<option className='options' value='irorun'>
							irorun
						</option>
					</select>
				</label>
				<label>
					Username
					<input type='text' value={username} onChange={(e) => setUsername(e.target.value)} placeholder='User' />
				</label>
				<label>
					Email
					<input type='email' value={email} onChange={(e) => setEmail(e.target.value)} required pattern='[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}' placeholder='Email' />
				</label>
				<label>
					Date
					<input type='date' value={date} onChange={(e) => setDate(e.target.value)} placeholder='Date' />
				</label>
				<label>
					Phone Number
					<input type='text' value={phoneNumber} onChange={(e) => setPhoneNumber(e.target.value)} placeholder='Phone' />
				</label>

				<label>
					Status
					<select value={status} onChange={(e) => setStatus(e.target.value)}>
						<option className='options' value='' disabled>
							Select
						</option>
						<option className='options' value='active'>
							Active
						</option>
						<option className='options' value='pending'>
							Pending
						</option>
						<option className='options' value='inactive'>
							Inactive
						</option>
						<option className='options' value='blacklisted'>
							Blacklisted
						</option>
					</select>
				</label>
				<div className='button-container'>
					<button onClick={closeModal} className='btn cancel_button'>
						Cancel
					</button>
					<button onClick={handleApply} className='btn apply_button'>
						Apply
					</button>
				</div>
			</div>
		</div>
	);
};

export default Modal;
