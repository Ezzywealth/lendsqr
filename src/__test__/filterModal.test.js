// Modal.test.js
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import FilterModal from '../components/users/Table/FilterModal'; // Adjust the path accordingly

jest.mock('../Redux/store.ts', () => ({
	...jest.requireActual('../Redux/store.ts'),
	useAppDispatch: jest.fn(),
}));

describe('Modal component', () => {
	const setShowFilterModalMock = jest.fn();

	const defaultProps = {
		setShowFilterModal: setShowFilterModalMock,
	};

	test('renders LoginForm component', () => {
		const { container } = render(<FilterModal {...defaultProps} />);
		expect(container).toMatchSnapshot();
	});

	it('renders modal content and handles apply button click', () => {
		render(<FilterModal {...defaultProps} />);

		// Check if modal content is rendered
		expect(screen.getByLabelText('Organization')).toBeInTheDocument();
		expect(screen.getByLabelText('Username')).toBeInTheDocument();
		expect(screen.getByLabelText('Email')).toBeInTheDocument();
		expect(screen.getByLabelText('Date')).toBeInTheDocument();
		expect(screen.getByLabelText('Phone Number')).toBeInTheDocument();
		expect(screen.getByLabelText('Status')).toBeInTheDocument();
		expect(screen.getByText('Cancel')).toBeInTheDocument();
		expect(screen.getByText('Apply')).toBeInTheDocument();
	});

	it('handles cancel button click', () => {
		render(<FilterModal {...defaultProps} />);

		// Click the Cancel button
		fireEvent.click(screen.getByText('Cancel'));

		// Check if the setShowFilterModalMock function was called
		expect(setShowFilterModalMock).toHaveBeenCalledWith(false);
	});
});
