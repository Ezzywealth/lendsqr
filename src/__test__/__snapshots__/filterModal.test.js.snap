// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders LoginForm component 1`] = `
<div>
  <div
    class="modal show"
  >
    <div
      class="modal_content"
    >
      <label>
        Organization
        <select>
          <option
            disabled=""
            value=""
          >
            Select
          </option>
          <option
            class="options"
            value="lendsqr"
          >
            lendsqr
          </option>
          <option
            class="options"
            value="lendstar"
          >
            lendstar
          </option>
          <option
            class="options"
            value="irorun"
          >
            irorun
          </option>
        </select>
      </label>
      <label>
        Username
        <input
          placeholder="User"
          type="text"
          value=""
        />
      </label>
      <label>
        Email
        <input
          pattern="[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\\\.[a-zA-Z]{2,}"
          placeholder="Email"
          required=""
          type="email"
          value=""
        />
      </label>
      <label>
        Date
        <input
          placeholder="Date"
          type="date"
          value=""
        />
      </label>
      <label>
        Phone Number
        <input
          placeholder="Phone"
          type="text"
          value=""
        />
      </label>
      <label>
        Status
        <select>
          <option
            class="options"
            disabled=""
            value=""
          >
            Select
          </option>
          <option
            class="options"
            value="active"
          >
            Active
          </option>
          <option
            class="options"
            value="pending"
          >
            Pending
          </option>
          <option
            class="options"
            value="inactive"
          >
            Inactive
          </option>
          <option
            class="options"
            value="blacklisted"
          >
            Blacklisted
          </option>
        </select>
      </label>
      <div
        class="button_container"
      >
        <button
          class="btn cancel_button"
        >
          Cancel
        </button>
        <button
          class="btn apply_button"
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</div>
`;
